front/src/common/Map/Search/MapSearchLine.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { useEffect, useState } from 'react';
+import InputSNCF from 'common/BootstrapSNCF/InputSNCF';
+import { useTranslation } from 'react-i18next';
+import { useDebounce } from 'utils/helpers';
+import { useSelector, useDispatch } from 'react-redux';
+import { getInfraID } from 'reducers/osrdconf/selectors';
+import { Viewport, updateLineSearchCode, updateMapSearchMarker } from 'reducers/map';
+import nextId from 'react-id-generator';
+import { BBox } from '@turf/helpers';
+import bbox from '@turf/bbox';
+import WebMercatorViewport from 'viewport-mercator-project';
+import { getMap } from 'reducers/map/selectors';
+import { Zone, osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { searchPayloadType } from '../const';
+import LineCard from './LineCard';
+import { ILineSearchResult } from './searchTypes';
+
+type MapSearchLineProps = {
+  updateExtViewport: (viewport: Partial<Viewport>) => void;
+};
+
+const MapSearchLine: React.FC<MapSearchLineProps> = ({ updateExtViewport }) => {
+  const infraID = useSelector(getInfraID);
+  const map = useSelector(getMap);
+  const { t } = useTranslation(['map-search']);
+  const dispatch = useDispatch();
+  const [postSearch] = osrdEditoastApi.usePostSearchMutation();
+  const [searchState, setSearchState] = useState<string>('');
+  const [dontSearch, setDontSearch] = useState(false);
+  const [searchResults, setSearchResults] = useState<ILineSearchResult[] | undefined>(undefined);
+  const [dataTrackZone, setDataTrackZone] = useState<{ id: number; lineCode: number }>({
+    id: 0,
+    lineCode: 0,
+  });
+  const [skip, setSkip] = useState<boolean>(true);
```

Comment:
I don't really understand what skip does

Reply:
skip is used to prevent editoast from being called (RTK request line 37) when the component is mounted

Reply:
Ok got it, thanks !! I'm not used to rtk